hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/jdbc/PredicatePushDownJdbcSqlConnectorTest.java

```diff
@@ -0,0 +1,174 @@
+
+                // Mathematical operators
+                "SELECT name FROM people WHERE age + 1 = 31",
+                "SELECT name FROM people WHERE age - 1 = 29",
+                "SELECT name FROM people WHERE age * 2 = 60",
+                "SELECT name FROM people WHERE age / 5 = 6",
+
+                // Conditional expressions
+                "SELECT name FROM people WHERE CASE age WHEN 30 THEN TRUE ELSE FALSE END",
+                "SELECT name FROM people WHERE CASE WHEN age = 30 THEN TRUE ELSE FALSE END",
+                "SELECT name FROM people WHERE NULLIF(age, 30) IS NULL",
+                "SELECT name FROM people WHERE COALESCE(nullable_column, nullable_column_reverse) = 'not null reverse'",
+
+                // Conversions
+                "SELECT name FROM people WHERE CAST(age AS VARCHAR) = '30'",
+
+                // String functions
+                "SELECT name FROM people WHERE 'Hello '||name = 'Hello John Doe'",
+                "SELECT name FROM people WHERE LENGTH(data) = 12",
+                "SELECT name FROM people WHERE LOWER(name) = 'john doe'",
+                "SELECT name FROM people WHERE UPPER(name) = 'JOHN DOE'",
+
+                // https://docs.hazelcast.com/hazelcast/5.2/sql/functions-and-operators#hide-nav
+                // Mathematical Functions
+                // String Functions
+                // Trigonometric Functions
+        };
+    }
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        initializePredicatePushDownTest(new H2DatabaseProvider());
+    }
+
+    protected static void initializePredicatePushDownTest(TestDatabaseProvider provider) throws SQLException {
+        initialize(provider);
+
+        tableName = "people";
+
+        createTable(tableName,
+                "id INT PRIMARY KEY",
+                "name VARCHAR(100)",
+                "age INT",
+                "data VARCHAR(100)",
+                "a BOOLEAN", "b BOOLEAN", "c BOOLEAN", "d BOOLEAN",
+                "nullable_column VARCHAR(100)",
+                "nullable_column_reverse VARCHAR(100)"
+        );
+
+        executeJdbc("INSERT INTO " + tableName + " VALUES (1, 'John Doe', 30, '{\"value\":42}', true, true, false, " +
+                "true, null, 'not null reverse')");
+        executeJdbc("INSERT INTO " + tableName + " VALUES (2, 'Jane Doe', 35, '{\"value\":0}', false, false, true, " +
+                "true, 'not null', null)");
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        // Needs to be in @Before because mappings are cleared between tests
+        sqlService.execute(
+                "CREATE MAPPING " + tableName + " DATA CONNECTION " + TEST_DATABASE_REF
+        );
+    }
+
+    @Test
+    public void runQueryExpectSingleResult_and_FullScanPhysicalRel() {
+        assertRowsAnyOrder(query, new Row("John Doe"));
+
+        assertExplain(
+                "EXPLAIN " + query,
```

Comment:
if I understand correctly, this is inteded to check if the plan consist only in FullScanPhysicalRel, without Calc. If so, we can also add explicit check for number of rows in plan = 1

Reply:
```
            assertThat(rows).containsExactly(rels);
```
here
https://github.com/hazelcast/hazelcast/pull/24521/files#diff-eebb57c32fcda5753e414657a2bc9a3fc615acfae2953f633d4e7c3d773d3fc5R171

checks for the size of `rows` is same as `rels`. Is that enough, or do you want explicit `assertThat(rows).hasSameSizeAs(rels);`?